test_openai_finetune.py

```diff
@@ -0,0 +1,245 @@
+
+    # Last, we can look at the results of the different formatting operations before proceeding with creating a fine-tuning job:
+
+    # Warnings and tokens counts
+    n_missing_system = 0
+    n_missing_user = 0
+    n_messages = []
+    convo_lens = []
+    assistant_message_lens = []
+
+    for ex in dataset:
+        messages = ex["messages"]
+        if not any(message["role"] == "system" for message in messages):
+            n_missing_system += 1
+        if not any(message["role"] == "user" for message in messages):
+            n_missing_user += 1
+        n_messages.append(len(messages))
+        convo_lens.append(num_tokens_from_messages(messages))
+        assistant_message_lens.append(num_assistant_tokens_from_messages(messages))
+
+    print("Num examples missing system message:", n_missing_system)
+    print("Num examples missing user message:", n_missing_user)
+    print_distribution(n_messages, "num_messages_per_example")
+    print_distribution(convo_lens, "num_total_tokens_per_example")
+    print_distribution(assistant_message_lens, "num_assistant_tokens_per_example")
+    n_too_long = sum(l > 4096 for l in convo_lens)
+    print(
+        f"\n{n_too_long} examples may be over the 4096 token limit, they will be truncated during fine-tuning"
+    )
+
+    # Pricing and default n_epochs estimate
+    MAX_TOKENS_PER_EXAMPLE = 4096
+
+    MIN_TARGET_EXAMPLES = 100
+    MAX_TARGET_EXAMPLES = 25000
+    TARGET_EPOCHS = 3
+    MIN_EPOCHS = 1
+    MAX_EPOCHS = 25
+
+    n_epochs = TARGET_EPOCHS
+    n_train_examples = len(dataset)
+    if n_train_examples * TARGET_EPOCHS < MIN_TARGET_EXAMPLES:
+        n_epochs = min(MAX_EPOCHS, MIN_TARGET_EXAMPLES // n_train_examples)
+    elif n_train_examples * TARGET_EPOCHS > MAX_TARGET_EXAMPLES:
+        n_epochs = max(MIN_EPOCHS, MAX_TARGET_EXAMPLES // n_train_examples)
+
+    n_billing_tokens_in_dataset = sum(
+        min(MAX_TOKENS_PER_EXAMPLE, length) for length in convo_lens
+    )
+    print(
+        f"Dataset has ~{n_billing_tokens_in_dataset} tokens that will be charged for during training"
+    )
+    print(f"By default, you'll train for {n_epochs} epochs on this dataset")
+    print(
+        f"By default, you'll be charged for ~{n_epochs * n_billing_tokens_in_dataset} tokens"
+    )
+    print("See pricing page to estimate total costs")
+
+
+openai_validate_data(train_file_path)
+
+
+# Upload the dataset
+openai_train_file_info = client.files.create(
+    file=open(train_file_path, "rb"), purpose="fine-tune"
+)
+
+# you may need to wait a couple of minutes for OpenAI to process the file
+print(openai_train_file_info)
+
+model = "gpt-3.5-turbo"
+n_epochs = 3
+
+openai_ft_job_info = client.fine_tuning.jobs.create(
+    training_file=openai_train_file_info.id,
+    model=model,
+    hyperparameters={"n_epochs": n_epochs},
+)
+
+ft_job_id = openai_ft_job_info.id
+
+print(openai_ft_job_info)
+
+WandbLogger.sync_job_id(id=ft_job_id)
```

Comment:
is it just me or I don't see a `sync_job_id` method in the logger?

Reply:
Hey this file will be removed. Sorry got commited by mistake.

Reply:
But yeah, the `sync_job_id`'s logic was moved to `sync`.